addons/dropbox/models.py

```diff
@@ -190,43 +192,38 @@ def get_folders(self, **kwargs):
                 }
             }]
 
-        client = DropboxClient(self.external_account.oauth_key)
-        file_not_found = HTTPError(http.NOT_FOUND, data={
-            'message_short': 'File not found',
-            'message_long': 'The Dropbox file you requested could not be found.'
-        })
-
-        max_retry_error = HTTPError(http.REQUEST_TIMEOUT, data={
-            'message_short': 'Request Timeout',
-            'message_long': 'Dropbox could not be reached at this time.'
-        })
+        client = Dropbox(self.external_account.oauth_key)
 
         try:
-            metadata = client.metadata(folder_id)
-        except ErrorResponse:
-            raise file_not_found
-        except MaxRetryError:
-            raise max_retry_error
-
-        # Raise error if folder was deleted
-        if metadata.get('is_deleted'):
-            raise file_not_found
+            folder_id = '' if folder_id == '/' else folder_id
+            list_folder = client.files_list_folder(folder_id)
+            contents = [x for x in list_folder.entries]
+            while list_folder.has_more:
```

Comment:
This function is only supposed  to return metadata about folders beneath an optional `folder_id` kwarg. Does the new DropBox client not have an way to get only folders, to limit the number of outgoing requests?